store/store.go

```diff
@@ -908,7 +940,13 @@ func (s *Store) fakeChannels(snapID string, user *auth.UserState) (map[string]*s
 		Data:        jsonData,
 	}
 
-	resp, err := s.retryRequest(context.TODO(), s.client, reqOptions, user)
+	var results struct {
+		Payload struct {
+			SnapDetails []*snapDetails `json:"clickindex:package"`
+		} `json:"_embedded"`
+	}
+
+	resp, err := s.retryRequestDecode(context.TODO(), s.client, reqOptions, user, decodeStrategy{decodeOnHTTP200, &results, nil, nil})
```

Comment:
Is there a reason why the approach we discussed does not work?  Compare the above with this:
```
resp, err := s.retryRequestJSON(context.TODO(), s.client, reqOptions, user, &results, nil)
```